controllers/quote.js

```diff
@@ -157,37 +173,103 @@ exports.get = function *() {
         'source transfers')
     }
   } else {
-    sourceExpiryDuration = destinationExpiryDuration +
-      config.expiry.minMessageWindow
+    sourceExpiryDuration = destinationExpiryDuration + config.expiry.minMessageWindow
   }
 
-  let paymentTemplate = {
+  this.destinationExpiryDuration = destinationExpiryDuration
+  this.sourceExpiryDuration = sourceExpiryDuration
+  this.source_amount = params.source_amount
+  this.destination_amount = params.destination_amount
+
+  this.source_ledger = params.source_ledger
+  this.destination_ledger = params.destination_ledger
+  this.source_account = params.source_account || null
+  this.destination_account = params.destination_account || null
+}
+
+QuoteQuery.prototype.loadLedgers = function * () {
+  if (this.source_ledger || this.source_account) {
+    this.source_ledger = this.source_ledger || (yield getAccountLedger(this.source_account))
+  } else {
+    throw new Error('Missing required parameter: source_ledger or source_account')
+  }
+  if (this.destination_ledger || this.destination_account) {
+    this.destination_ledger = this.destination_ledger || (yield getAccountLedger(this.destination_account))
+  } else {
+    throw new Error('Missing required parameter: destination_ledger or destination_account')
+  }
+}
+
+QuoteQuery.prototype.toQuoteArgs = function () {
```

Comment:
This isn't a generic JSON representation of the `QuoteQuery` -- if it was, it would include expiries, accounts, etc. This is a specific object -- representing arguments for the `getQuote` function, hence `toQuoteArgs`.